traits/has_traits.py

```diff
@@ -419,11 +419,14 @@ def update_traits_class_dict(class_name, bases, class_dict):
 
         # Warn name collisions.
         if check_name_collision and _is_disallowed_prefix(name):
-            warnings.warn("The attribute named '{}' of class {} uses one of "
-                          "the reserved prefixes: {}. Consider renaming it."
-                          .format(name, class_name,
-                                  ",".join(DisallowedNamePrefixes)),
-                          UserWarning, 3)
+            warnings.warn(
+                "The attribute named {!r} of class {} uses one of "
+                "the reserved prefixes: {}. Consider renaming it.".format(
+                    name, class_name,
+                    ", ".join(
+                        repr(prefix) for prefix in DisallowedNamePrefixes)
+                ),
+                UserWarning, 3)
```

Comment:
While we're here, could we add names for these arguments? The `UserWarning` is fairly clear, but the `3` would be hard to guess for someone reading the code. `stacklevel=3` would be clearer.